From 3f5b7fca28964dd8fa157ba4ddac91f1f0f8cd08 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:50 -0400
Subject: [fs] nfsd: fix test_stateid error reply encoding

Message-id: <1403560314-32525-49-git-send-email-bfields@redhat.com>
Patchwork-id: 84476
O-Subject: [Patch RHEL7 048/172] nfsd4: fix test_stateid error reply encoding
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

If the entire operation fails then there's nothing to encode.

Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: a11fcce1544df08c723d950ff0edef3adac40405
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index de7bc8f..aa04a6a 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -3473,6 +3473,9 @@ nfsd4_encode_test_stateid(struct nfsd4_compoundres *resp, __be32 nfserr,
  struct nfsd4_test_stateid_id *stateid, *next;
  __be32 *p;
 
+ if (nfserr)
+  return nfserr;
+
  RESERVE_SPACE(4 + (4 * test_stateid->ts_num_ids));
  *p++ = htonl(test_stateid->ts_num_ids);
 
-- 
1.7.1